axiom/monitors_test.go

```diff
@@ -215,3 +215,185 @@ func TestMonitorsService_Delete(t *testing.T) {
 	err := client.Monitors.Delete(context.Background(), "testID")
 	require.NoError(t, err)
 }
+
+func TestMonitorsService_AdvancedUsage(t *testing.T) {
```

Comment:
Any reason these tests can't be folded into the `CreateMonitor` test?

The unit tests are there to make sure all params/fields/etc. are correctly passed to the server and back and that all the marshallers, etc. are involved correctly. There is no harm in having a single test that just populates **all the fields** for a create monitor request, even if that might be invalid when testing against a real environment. But in unit tests, everything is mocked, anyways.

I would have expected to do the "advanced testing" and more behavioural oriented tests in the integration test suite for monitors.